operator-common/src/main/java/io/strimzi/operator/common/operator/resource/ConfigMapOperator.java

```diff
@@ -36,6 +40,19 @@ public ConfigMapOperator(Vertx vertx, KubernetesClient client) {
 
     @Override
     protected Future<ReconcileResult<ConfigMap>> internalPatch(String namespace, String name, ConfigMap current, ConfigMap desired) {
+        JsonNode diff = JsonDiff.asJson(patchMapper().valueToTree(current), patchMapper().valueToTree(desired));
+        boolean metricsChanged = false;
+        for (JsonNode d : diff) {
+            if (d.get("path").asText().equals("/data/metrics-config.yml") && d.get("op").asText().equals("replace")) {
+                metricsChanged = true;
+            }
+        }
+
+        if (diff.size() == 1 && metricsChanged) {
```

Comment:
What if there are more than one diff but metrics changed? I mean does we need the first check on the size? Just asking...

Reply:
If there is more then one change, we want to trigger roll update because of that change which is not in the metrics settings.